src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/CallSiteChain.cs

```diff
@@ -47,9 +47,12 @@ private string CreateCircularDependencyExceptionMessage(Type type)
             return messageBuilder.ToString();
         }
 
-        private void AppendResolutionPath(StringBuilder builder, Type currentlyResolving = null)
+        private void AppendResolutionPath(StringBuilder builder, Type currentlyResolving)
         {
-            foreach (KeyValuePair<Type, ChainItemInfo> pair in _callSiteChain.OrderBy(p => p.Value.Order))
+            var ordered = new List<KeyValuePair<Type, ChainItemInfo>>(_callSiteChain);
+            ordered.Sort((a, b) => a.Value.Order.CompareTo(b.Value.Order));
```

Comment:
FWIW, this could avoid the List and just use an array, e.g.
```C#
var ordered = new KeyValuePair<Type, ChainItemInfo>[_callSiteChain.Length];
((ICollection<KeyValuePair<TKey, TValue>>)_callSiteChain).CopyTo(ordered, 0);
Array.Sort(ordered, (a, b) => a.Value.Order.CompareTo(b.Value.Order));
```
That'll also let the compiler change the subsequent foreach into a for loop.

Regardless, nice to see the unnecessary OrderBy removal.